crates/e2e/tests/e2e/eth_flow.rs

```diff
@@ -138,37 +155,57 @@ async fn eth_flow_tx(web3: Web3) {
 
     sumbit_order(&ethflow_order, &trader, &contracts).await;
 
-    // Automatically pick up onchain order without any API request
-    maintenance.run_maintenance().await.unwrap();
-
     test_order_availability_in_api(&client, &ethflow_order, &trader.address(), &contracts).await;
 
-    let mut driver =
-        setup_naive_solver_uniswapv2_driver(&web3, &contracts, base_tokens, block_stream, solver)
-            .await;
-    driver.single_run().await.unwrap();
-
-    // Update orderbook status
-    maintenance.run_maintenance().await.unwrap();
-    solvable_orders_cache.update(0).await.unwrap();
-
+    tracing::info!("waiting for trade");
+    wait_for_condition(Duration::from_secs(10), || async {
+        solvable_orders().await.unwrap() == 1
+    })
+    .await
+    .unwrap();
+    crate::services::start_old_driver(&contracts, &SOLVER_PK, &[]);
     test_order_was_settled(&ethflow_order, &web3).await;
 
     test_trade_availability_in_api(&client, &ethflow_order, &trader.address(), &contracts).await;
 }
 
 async fn eth_flow_indexing_after_refund(web3: Web3) {
     shared::tracing::initialize_reentrant(
-        "warn,orderbook=debug,orderbook=debug,solver=debug,autopilot=debug",
+        "e2e=debug,orderbook=debug,solver=debug,autopilot=debug,\
+         orderbook::api::request_summary=off",
     );
     shared::exit_process_on_panic::set_panic_hook();
+
+    crate::services::clear_database().await;
     let contracts = crate::deploy::deploy(&web3).await.expect("deploy");
 
-    let mut accounts = AccountAssigner::new(&web3).await;
-    let trader = accounts.assign_free_account();
-    let dummy_trader = accounts.assign_free_account();
-    let refunder = accounts.assign_free_account();
-    let solver = accounts.default_deployer; // is a solver by default
+    const SOLVER_PK: [u8; 32] =
+        hex!("0000000000000000000000000000000000000000000000000000000000000001");
+    let solver = Account::Offline(PrivateKey::from_raw(SOLVER_PK).unwrap(), None);
+    contracts
+        .gp_authenticator
+        .add_solver(solver.address())
+        .send()
+        .await
+        .unwrap();
+
+    const REFUNDER_PK: [u8; 32] =
+        hex!("0000000000000000000000000000000000000000000000000000000000000002");
+    let refunder = Account::Offline(PrivateKey::from_raw(TRADER_PK).unwrap(), None);
+    const TRADER_PK: [u8; 32] =
+        hex!("0000000000000000000000000000000000000000000000000000000000000003");
+    let trader = Account::Offline(PrivateKey::from_raw(TRADER_PK).unwrap(), None);
+    const DUMMY_TRADER_PK: [u8; 32] =
+        hex!("0000000000000000000000000000000000000000000000000000000000000004");
+    let dummy_trader = Account::Offline(PrivateKey::from_raw(TRADER_PK).unwrap(), None);
```

Comment:
Should all those accounts actually be initialized with the same private key?
Also TIL you can use a `const` variable before the line initializing it.

Reply:
Looks like copy-pasta!

Reply:
fixed